src/scripts/main.js

```diff
@@ -1,5 +1,34 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const wall = document.querySelector('.wall');
+
 document.addEventListener('click', e => {
   // write code here
+  function getSize(el, prop) {
+    return parseFloat(
+      window.getComputedStyle(el)[prop]
+    );
+  };
+
+  if (e.target.className === 'wall') {
+    const spiderWidth = getSize(spider, 'width');
+    const spiderHeight = getSize(spider, 'height');
+    const wallWidth = getSize(wall, 'width');
+    const wallHeight = getSize(wall, 'height');
+    const halfSpiderWidth = spiderWidth / 2;
+    const halfSpiderHeight = spiderHeight / 2;
+
+    const newX = (e.offsetX < halfSpiderWidth)
+      ? 0 : (e.offsetX > wallWidth - halfSpiderWidth)
+        ? wallWidth - spiderWidth : e.offsetX - halfSpiderWidth;
```

Comment:
Nested ternary operators are not the best idea. Your code looks a bit complicated

Reply:
![image](https://user-images.githubusercontent.com/51528776/211827407-f5128358-6aef-4af3-8f5f-06cc8bde943c.png)

Reply:
i can use switch also))